src/lib/maintenance/db.js

```diff
@@ -0,0 +1,347 @@
+      action = dailyMaintenance(offset, limit);
+      break;
+    default:
+      // Throw an error if an invalid maintenance type is provided.
+      throw new Error(`Invalid DB maintenance type: ${type}`);
+  }
+
+  return action; // Return the maintenance action.
+};
+
+/**
+ * Adds a job to the maintenance queue for database maintenance.
+ * @param {string} type - The type of database maintenance to perform.
+ * @param {*} data - Optional data to be included with the maintenance job.
+ * @param {number} percent - Optional percentage value.
+ * @returns {void}
+ * @throws {Error} If an error occurs while adding the job to the maintenance queue.
+ */
+const enqueueDBMaintenanceJob = async (
+  type,
+  data,
+  percent = null, // optional parameter with default value of null
+) => enqueueMaintenanceJob(
+  MAINTENANCE_CATEGORY.DB, // constant representing the category of maintenance
+  {
+    type, // shorthand property notation for type: type
+    ...(!data // spread operator used to merge properties of two objects
+      // if data is not provided, call nextBlock function and merge its result
+      ? await nextBlock(type, percent)
+      : data), // otherwise, merge the provided data object
+  },
+);
+
+// This code adds a queue processor for database maintenance tasks.
+// The MAINTENANCE_CATEGORY.DB is used to identify the category of maintenance task.
+// The dbMaintenance function is passed as the callback function to be executed when
+// a task in this category is processed.
+addQueueProcessor(MAINTENANCE_CATEGORY.DB, dbMaintenance);
+
+// Adds a cron job with the specified maintenance category, type, and function to execute
+addCronJob(
+  MAINTENANCE_CATEGORY.DB, // The maintenance category is "DB"
+  MAINTENANCE_TYPE.DAILY_DB_MAINTENANCE, // The maintenance type is "DAILY_DB_MAINTENANCE"
+  // The function to execute takes in the category, type, timezone, and schedule parameters
+  (category, type, timezone, schedule) => new CronJob(
+    schedule, // The schedule parameter specifies when the job should run
+    () => enqueueDBMaintenanceJob( // Enqueues a database maintenance job
+      MAINTENANCE_TYPE.DAILY_DB_MAINTENANCE, // The maintenance type is "DAILY_DB_MAINTENANCE"
+      null, // no extra data passed
+      0.2, // Only 20% of the tables will be maintained each day
+    ),
+    null,
+    true,
+    timezone, // The timezone parameter specifies the timezone in which the job should run
+  ),
+  /**
+   * This cron expression breaks down as follows:
+   *  0 - The minute when the job will run (in this case, 0 minutes past the hour)
+   *  23 - The hour when the job will run (in this case, 11 pm)
+   *  * - The day of the month when the job will run (in this case, any day of the month)
+   *  * - The month when the job will run (in this case, any month)
+   *  * - The day of the week when the job will run (in this case, any day of the week)
+   * */
+  '0 23 * * *',
```

Comment:
I modified this locally to */5 * * * * and restarted my local environment. Did not see it run after about 15 minutes. Is there something else I'd have to do?

Reply:
in cron.js you need to comment out / remove the check for production only for cron jobs also.

Reply:
Cool thanks. Works as expected. The logs do say "Successfully performed DB maintenance for undefined DAILY DB MAINTENANCE {}" so I think maybe something isn't getting passed in.